test(network): added unit test for loadOrCreateKey

LeonLow97 opened the pull request:
## Description

> Added unit test for loadOrCreateKey, tested with the following cases:

1. Valid Private Key created
2. Invalid Private Key in existing file
3. Input supplied with directory instead of file path
4. Input supplied with invalid file path

## Related Issue

> Nil

Fixes # (issue)

- Increased Unit Test Coverage from 76.6% to 82.1% in `network_test.go`

LeonLow97 commented:
Resolves #1766

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+func TestLoadOrCreateKey(t *testing.T) {
+	t.Run("ReturnsSuccessfullyWhenValidKeyCreated", func(t *testing.T) {
+		tempDir := t.TempDir()
+		keyPath := filepath.Join(tempDir, "valid_key")
```
Use [utils.TempFilePath](https://github.com/pactus-project/pactus/blob/6b43f89771b5785bbb4507da04a4cd44e5a17f84/util/io.go#L72) instead.

LeonLow97 replied:
I have changed this to use `util.TempFilePath`

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
 }
+
+func TestLoadOrCreateKey(t *testing.T) {
+	t.Run("ReturnsSuccessfullyWhenValidKeyCreated", func(t *testing.T) {
```
Write proper description for tests. Here for example: "Should create network key if it doesn't exist."

LeonLow97 replied:
I have changed all unit test titles to follow this format

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+		previousValidKey, err := loadOrCreateKey(keyPath)
+		assert.NoError(t, err)
+
+		// Compare public keys of both private keys
```
avoiding over explaining in comments

LeonLow97 replied:
I have removed this comment

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+		assert.NoError(t, err)
+
+		// Compare public keys of both private keys
+		assert.True(t, validKey.GetPublic().Equals(previousValidKey.GetPublic()))
```
user `assert.Equal(t, foo, bar)`

LeonLow97 replied:
I have changed this to use `assert.Equal` instead of `assert.True`

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+	t.Run("ReturnsErrorWhenInvalidKeyCreated", func(t *testing.T) {
+		tempFile, err := os.CreateTemp("", "invalid_key")
+		assert.NoError(t, err)
+		defer os.Remove(tempFile.Name())
```
No need to remove Temp file.. Keep test simple

LeonLow97 replied:
Good point, I have removed the `defer` statement

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+		defer os.Remove(tempFile.Name())
+
+		// Writes an invalid private key to the file, decoding key will fail later
+		err = os.WriteFile(tempFile.Name(), []byte("invalid_data"), 0644)
```
Use `utils.WriteFile`

LeonLow97 replied:
I have changed this to use `util.WriteFile`

b00f commented on network/network_test.go:
```diff
@@ -494,3 +496,48 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+	})
+
+	t.Run("ReturnsErrorWithInvalidFilePath", func(t *testing.T) {
+		key, err := loadOrCreateKey("/invalid-file-path")
```
Invalid directory in different OS has different pattern.

LeonLow97 replied:
I have changed this to use `invalidPath := string([]byte{0x00})` which is a null byte is invalid in file names across all OS

b00f reviewed (changes requested):
The most important test is missing. We want to ensure that the network key is properly loaded with the same network ID

LeonLow97 commented:
Hi @b00f , thanks for the informative review, I have addressed the comments.

<br>

> The most important test is missing. We want to ensure that the network key is properly loaded with the same network ID

Regarding this point, I have already included it in the first unit test, i.e., `Should load same network key after being created` where I first create a private key and then load the private key again. I compared both public keys to ensure that they are the same within the same file path. Let me know if I am missing anything here 👍

b00f commented on network/network_test.go:
```diff
@@ -494,3 +495,46 @@ func checkConnection(t *testing.T, networkP, networkB *network) {
+	})
+
+	t.Run("Should return error when file contains invalid private key", func(t *testing.T) {
+		tempFile, err := os.CreateTemp("", "invalid_key")
```
Use utils functions here: `keyPath := util.TempFilePath()`

LeonLow97 replied:
I have changed this to use `util.TempFilePath`

b00f commented:
Thanks @LeonLow97 Only one comment left. Once it resolved we can merge the PR.

LeonLow97 commented:
Awesome @b00f , thanks for the review, I have made the relevant changes 👍

LeonLow97 commented:
Unit test seems to be failing on Windows OS that is unrelated to my change, probably an environment configuration issue

```
Running TestNetwork/Reachability_Status
--- FAIL: TestNetwork (14.47s)
    network_test.go:135: TestNetwork seed is 1747036069997689600
    --- FAIL: TestNetwork/Reachability (1.46s)
        network_test.go:256: 
            	Error Trace:	D:/a/pactus/pactus/network/network_test.go:256
            	Error:      	Condition never satisfied
            	Test:       	TestNetwork/Reachability
07:48:04 WRN error on sending msg error="libp2p error: connection failed" _network="<peer.ID 12*3rgoSv> - TestNetwork: {3}" pid=12D3KooWSybER2xyCMyh69tp2kmGidS6Pq2biksMErTgzZ4vQb5F
```
